domain/account/usecase/deposit.go

```diff
@@ -1,29 +1,27 @@
 package usecase
 
-import (
-	"errors"
-)
-
-func (c AccountUc) Deposit(CPF string, amount float64) error {
+import "github.com/francisleide/ChallengeGo/domain/entities"
 
+func (c AccountUc) Deposit(CPF string, amount float64) (entities.TransactionOutput, error) {
+	var depositOutput entities.TransactionOutput
 	account, err := c.r.FindOne(CPF)
+	depositOutput.PreviousBalance = account.Balance
+	depositOutput.ID = account.ID
 	if err != nil {
-		c.log.WithError(err).Errorln("failed to retrieve account")
-		//TODO add a new sentinel
-		return errors.New("failed to retrieve account")
+		c.log.WithError(err).Errorln(ErrorRetrieveAccount)
+		return entities.TransactionOutput{}, ErrorRetrieveAccount
```

Comment:
here you're returning and logging the error at the same level, which should be avoided. take a look at https://web3.coach/golang-how-to-handle-errors-five-rules

Reply:
The post is excellent, it helped a lot!